### PR TITLE
Add useStringAsKey flag.

### PR DESCRIPTION
**Pull request by gskrobisz**

### CODE REVIEW

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala [hunk 1]
```diff
@@ -34,8 +34,8 @@ trait ConfluentKafkaAvroDeserializerFactory extends LazyLogging {
   }
 }
 
-class ConfluentKafkaAvroDeserializationSchemaFactory(schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory)
-  extends KafkaAvroValueDeserializationSchemaFactory with ConfluentKafkaAvroDeserializerFactory {
+class ConfluentKafkaAvroDeserializationSchemaFactory(schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory, _useStringAsKey: Boolean)
```
Comment (arkadius) [inline engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala:37]: Can you name this parameter without underscore? You don't have variable shadowing here so it is not necessary

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/serialization/KafkaAvroDeserializationSchemaFactory.scala [hunk 2]
```diff
@@ -39,7 +37,8 @@ trait KafkaAvroDeserializationSchemaFactory extends Serializable {
   * Abstract base implementation of [[KafkaAvroDeserializationSchemaFactory]]
   * which uses Kafka's Deserializer in returned Flink's KeyedDeserializationSchema. It deserializes only value.
   */
-abstract class KafkaAvroValueDeserializationSchemaFactory
+// TODO: remove class and use only KeyValue deserialization with "_useStringAsKey" flag.
+abstract class KafkaAvroValueDeserializationSchemaFactory(_useStringAsKey: Boolean)
```
Comment (arkadius) [inline engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/serialization/KafkaAvroDeserializationSchemaFactory.scala:41]: The same here

**Review by arkadius — APPROVED** [review]

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala (hunk 1 above)
Comment (gskrobisz) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala]: Fixed.

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/serialization/KafkaAvroDeserializationSchemaFactory.scala (hunk 2 above)
Comment (gskrobisz) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/serialization/KafkaAvroDeserializationSchemaFactory.scala]: Fixed.

**Pull request merged**